libvast/src/system/index.cpp

```diff
@@ -654,16 +654,15 @@ index(index_actor::stateful_pointer<index_state> self,
         // importer.
         self->send_exit(self, err);
       }
-      VAST_DEBUG("index finalized streaming");
     });
   self->set_exit_handler([self](const caf::exit_msg& msg) {
     VAST_DEBUG("{} received EXIT from {} with reason: {}", self, msg.source,
                msg.reason);
     // Flush buffered batches and end stream.
+    self->state.stage->shutdown(); // closes inbound paths
     self->state.stage->out().fan_out_flush();
+    self->state.stage->out().close(); // closes outbound paths
```

Comment:
It would be good if we had a better understanding of what these do and why the order matters.
Maybe it also makes sense to put this sequence into it's own function.

Reply:
Yes, good point. I actually think the final `force_emit_batches()` is not needed at all, since caf should automatically try to force underfull batches after the `max-batch-delay` of 5ms has run out.

Reply:
On second thought, I think it would be better to make this refactoring in a separate topic branch: A function in full generality would add an additional call to `shutdown()` to some places, and while I think it is correct it will give me additional peace of mind to know that this can be reverted independently in case it breaks anything.